support-business/src/main/java/br/com/support/business/facade/impl/ClientFacade.java

```diff
@@ -34,13 +35,13 @@
 	private ClientMapper clientMapper;
 
 	@Override
-	public void saveClient(ClientDTO clientDTO) throws ExistingEntityException {
-		//TODO check if the client already exists
-		Client client = null;
+	public ClientDTO saveClient(ClientDTO clientDTO) throws ExistingEntityException {
+		Example<Client> example = Example.of(this.clientMapper.map(clientDTO));
```

Comment:
Wouldn't be necessary to define a ExampleMatcher to specify what attributes should be taken into account? Ex.: Will a client with the same cpf and different name return?

Reply:
The Example will contain all filled properties, for example, if the client in the Example has name and cpf, both will be used for matching. Talking about to use or not, I agree with you, it's better to use.